core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/constructionheuristic/placer/QueuedValuePlacerFactory.java

```diff
@@ -31,9 +31,7 @@ public QueuedValuePlacerFactory(QueuedValuePlacerConfig placerConfig) {
 
     @Override
     public QueuedValuePlacer<Solution_> buildEntityPlacer(HeuristicConfigPolicy<Solution_> configPolicy) {
-        EntityDescriptor<Solution_> entityDescriptor =
-                config.getEntityClass() == null ? deduceEntityDescriptor(configPolicy.getSolutionDescriptor())
-                        : deduceEntityDescriptor(configPolicy.getSolutionDescriptor(), config.getEntityClass());
+        EntityDescriptor<Solution_> entityDescriptor = deduceEntityDescriptor(configPolicy, config.getEntityClass());
```

Comment:
I'm thinking out loud... Is it worth having the deduceEntityDescriptor method?
If we keep the null check here, we can call the two new methods directly.
Not sure if that is better - probably if the null check would proliferate, then keeping it inside the extra method is better.